python/lsst/ip/isr/isrStatistics.py

```diff
@@ -350,90 +354,136 @@ def measureCti(self, inputExp, overscans, gains):
+                             gains, isTrimmed=False):
+                for amp in detector.getAmplifiers():
+                    ampStats = {}
+                    readoutCorner = amp.getReadoutCorner()
+
+                    # Full data region.
+                    dataRegion = image[amp.getBBox()]
+                    serialOverscanImage = untrimmedImage[amp.getRawSerialOverscanBBox()]
+                    parallelOverscanImage = untrimmedImage[amp.getRawSerialOverscanBBox()]
+
+                    # Get the mean of the image
+                    ampStats["IMAGE_MEAN"] = afwMath.makeStatistics(dataRegion, self.statType,
+                                                                    self.statControl).getValue()
+
+                    # First and last image columns.
+                    colA = afwMath.makeStatistics(
+                        dataRegion.array[:, 0],
+                        self.statType,
+                        self.statControl,
+                    ).getValue()
+                    colZ = afwMath.makeStatistics(
+                        dataRegion.array[:, -1],
+                        self.statType,
+                        self.statControl,
+                    ).getValue()
+
+                    # First and last image rows.
+                    rowA = afwMath.makeStatistics(
+                        dataRegion.array[0, :],
+                        self.statType,
+                        self.statControl,
+                    ).getValue()
+                    rowZ = afwMath.makeStatistics(
+                        dataRegion.array[-1, :],
+                        self.statType,
+                        self.statControl,
+                    ).getValue()
 
+                    # We want these relative to the readout corner.  If that's
+                    # on the right side, we need to swap them.
+                    if readoutCorner in (ReadoutCorner.LR, ReadoutCorner.UR):
+                        ampStats["FIRST_COLUMN_MEAN"] = colZ
+                        ampStats["LAST_COLUMN_MEAN"] = colA
+                    else:
+                        ampStats["FIRST_COLUMN_MEAN"] = colA
+                        ampStats["LAST_COLUMN_MEAN"] = colZ
+
+                    # We want these relative to the readout corner.  If that's
+                    # on the top, we need to swap them.
+                    if readoutCorner in (ReadoutCorner.UR, ReadoutCorner.UL):
+                        ampStats["FIRST_ROW_MEAN"] = rowZ
+                        ampStats["LAST_ROW_MEAN"] = rowA
+                    else:
+                        ampStats["FIRST_ROW_MEAN"] = rowA
+                        ampStats["LAST_ROW_MEAN"] = rowZ
+
+                    # Measure the columns of the serial overscan and the rows
+                    # of the parallel overscan.
+                    nSerialOverscanCols = amp.getRawSerialOverscanBBox().getWidth()
+                    nParallelOverscanRows = amp.getRawParallelOverscanBBox().getHeight()
```

Comment:
Note that these are only going to work with an "LSST style" camera, which is fine, that's what we support.

Reply:
(The problem is that a different camera could have a different row/col orientation and width/height would therefore be reversed).